### ISSUE TITLE
Redis crash on FT.SEARCH (IR_GetCriteriaTester)

### ISSUE THREAD

**Issue by dme-development** [RediSearch/RediSearch#613]
We've experienced issues with Redis crashing on a FT.SEARCH command. Please find the full crash log below for the detailled error description and the FT.SEARCH command. 

This error behaviour is seen on the latest version of RediSearch compiled from the master branch.
It is not seen (and thus working as expected) on the version from tags/v1.4.3, so it might be a new issue committed to the master branch recently.

Crash log:

```
> 479:C 20 Feb 14:53:20.279 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
> 479:C 20 Feb 14:53:20.280 # Redis version=4.0.11, bits=64, commit=00000000, modified=0, pid=479, just started
> 479:C 20 Feb 14:53:20.280 # Configuration loaded
> 508:M 20 Feb 14:53:20.281 * Increased maximum number of open files to 10032 (it was originally set to 1024).
>                 _._                                                  
>            _.-``__ ''-._                                             
>       _.-``    `.  `_.  ''-._           Redis 4.0.11 (00000000/0) 64 bit
>   .-`` .-```.  ```\/    _.,_ ''-._                                   
>  (    '      ,       .-`  | `,    )     Running in standalone mode
>  |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
>  |    `-._   `._    /     _.-'    |     PID: 508
>   `-._    `-._  `-./  _.-'    _.-'                                   
>  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
>  |    `-._`-._        _.-'_.-'    |           http://redis.io        
>   `-._    `-._`-.__.-'_.-'    _.-'                                   
>  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
>  |    `-._`-._        _.-'_.-'    |                                  
>   `-._    `-._`-.__.-'_.-'    _.-'                                   
>       `-._    `-.__.-'    _.-'                                       
>           `-._        _.-'                                           
>               `-.__.-'                                               
> 
> 508:M 20 Feb 14:53:20.384 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
> 508:M 20 Feb 14:53:20.384 # Server initialized
> 508:M 20 Feb 14:53:20.384 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
> 508:M 20 Feb 14:53:20.384 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
> 508:M 20 Feb 14:53:20.579 * <ft> Low level api version 1 initialized successfully
> 508:M 20 Feb 14:53:20.579 * <ft> RediSearch version 1.4.2 (Git=v1.4.2-255-gf947fd74)
> 508:M 20 Feb 14:53:20.579 * <ft> concurrency: ON, gc: ON, prefix min length: 2, prefix max expansions: 200, query timeout (ms): 500, timeout policy: return, cursor read size: 1000, cursor max idle (ms): 300000, max doctable size: 1000000, search pool size: 20, index pool size: 8, 
> 508:M 20 Feb 14:53:20.817 * <ft> Initialized thread pool!
> 508:M 20 Feb 14:53:20.817 * Module 'ft' loaded from /home/redisuser01/RediSearch/build/redisearch.so
> 508:M 20 Feb 14:53:21.438 * DB loaded from disk: 0.621 seconds
> 508:M 20 Feb 14:53:21.438 * Ready to accept connections
> 
> 
> === REDIS BUG REPORT START: Cut & paste starting from here ===
> 508:M 20 Feb 14:54:05.440 # Redis 4.0.11 crashed by signal: 11
> 508:M 20 Feb 14:54:05.440 # Crashed running the instruction at: 0x7f4718223e7d
> 508:M 20 Feb 14:54:05.440 # Accessing address: 0x28
> 508:M 20 Feb 14:54:05.440 # Failed assertion: <no assertion failed> (<no file>:0)
> 
> ------ STACK TRACE ------
> EIP:
> /home/redisuser01/RediSearch/build/redisearch.so(IR_GetCriteriaTester+0x4d)[0x7f4718223e7d]
> 
> Backtrace:
> /usr/local/bin/redis-server 10.25.1.132:6379(logStackTrace+0x37)[0x55a62f27ac97]
> /usr/local/bin/redis-server 10.25.1.132:6379(sigsegvHandler+0xb0)[0x55a62f27b3c0]
> /lib/x86_64-linux-gnu/libpthread.so.0(+0x110e0)[0x7f471a5fd0e0]
> /home/redisuser01/RediSearch/build/redisearch.so(IR_GetCriteriaTester+0x4d)[0x7f4718223e7d]
> /home/redisuser01/RediSearch/build/redisearch.so(+0x4fc56)[0x7f471821fc56]
> /home/redisuser01/RediSearch/build/redisearch.so(NewUnionIterator+0x1b2)[0x7f4718220c02]
> /home/redisuser01/RediSearch/build/redisearch.so(Query_EvalNode+0x4d6)[0x7f471822dbc6]
> /home/redisuser01/RediSearch/build/redisearch.so(Query_EvalNode+0x33a)[0x7f471822da2a]
> /home/redisuser01/RediSearch/build/redisearch.so(QAST_Iterate+0x4a)[0x7f471822dfca]
> /home/redisuser01/RediSearch/build/redisearch.so(AREQ_ApplyContext+0x311)[0x7f4718240a51]
> /home/redisuser01/RediSearch/build/redisearch.so(+0x6d2b8)[0x7f471823d2b8]
> /home/redisuser01/RediSearch/build/redisearch.so(+0x6d4f9)[0x7f471823d4f9]
> /usr/local/bin/redis-server 10.25.1.132:6379(RedisModuleCommandDispatcher+0x43)[0x55a62f2a4553]
> /usr/local/bin/redis-server 10.25.1.132:6379(call+0x9e)[0x55a62f23b30e]
> /usr/local/bin/redis-server 10.25.1.132:6379(processCommand+0x4a7)[0x55a62f23bb07]
> /usr/local/bin/redis-server 10.25.1.132:6379(processInputBuffer+0x131)[0x55a62f24b571]
> /usr/local/bin/redis-server 10.25.1.132:6379(aeProcessEvents+0x2a0)[0x55a62f235780]
> /usr/local/bin/redis-server 10.25.1.132:6379(aeMain+0x2b)[0x55a62f235a1b]
> /usr/local/bin/redis-server 10.25.1.132:6379(main+0x4d3)[0x55a62f232673]
> /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1)[0x7f471a26d2e1]
> /usr/local/bin/redis-server 10.25.1.132:6379(_start+0x2a)[0x55a62f23297a]
> 
> ------ INFO OUTPUT ------
> # Server
> redis_version:4.0.11
> redis_git_sha1:00000000
> redis_git_dirty:0
> redis_build_id:9f4688a2298a3d79
> redis_mode:standalone
> os:Linux 4.9.0-8-amd64 x86_64
> arch_bits:64
> multiplexing_api:epoll
> atomicvar_api:atomic-builtin
> gcc_version:6.3.0
> process_id:508
> run_id:c2e3038516ce2e081631222cfe7617437765988f
> tcp_port:6379
> uptime_in_seconds:45
> uptime_in_days:0
> hz:10
> lru_clock:7166973
> executable:/usr/local/bin/redis-server
> config_file:/etc/redis/6379.conf
> 
> # Clients
> connected_clients:11
> client_longest_output_list:0
> client_biggest_input_buf:0
> blocked_clients:0
> 
> # Memory
> used_memory:280429016
> used_memory_human:267.44M
> used_memory_rss:311918592
> used_memory_rss_human:297.47M
> used_memory_peak:280969528
> used_memory_peak_human:267.95M
> used_memory_peak_perc:99.81%
> used_memory_overhead:5724514
> used_memory_startup:786736
> used_memory_dataset:274704502
> used_memory_dataset_perc:98.23%
> total_system_memory:8375959552
> total_system_memory_human:7.80G
> used_memory_lua:43008
> used_memory_lua_human:42.00K
> maxmemory:0
> maxmemory_human:0B
> maxmemory_policy:noeviction
> mem_fragmentation_ratio:1.11
> mem_allocator:jemalloc-4.0.3
> active_defrag_running:0
> lazyfree_pending_objects:0
> 
> # Persistence
> loading:0
> rdb_changes_since_last_save:0
> rdb_bgsave_in_progress:0
> rdb_last_save_time:1550670800
> rdb_last_bgsave_status:ok
> rdb_last_bgsave_time_sec:-1
> rdb_current_bgsave_time_sec:-1
> rdb_last_cow_size:0
> aof_enabled:0
> aof_rewrite_in_progress:0
> aof_rewrite_scheduled:0
> aof_last_rewrite_time_sec:-1
> aof_current_rewrite_time_sec:-1
> aof_last_bgrewrite_status:ok
> aof_last_write_status:ok
> aof_last_cow_size:0
> 
> # Stats
> total_connections_received:21
> total_commands_processed:157
> instantaneous_ops_per_sec:5
> total_net_input_bytes:18482
> total_net_output_bytes:3128619
> instantaneous_input_kbps:1.35
> instantaneous_output_kbps:37.47
> rejected_connections:0
> sync_full:0
> sync_partial_ok:0
> sync_partial_err:0
> expired_keys:0
> expired_stale_perc:0.00
> expired_time_cap_reached_count:0
> evicted_keys:0
> keyspace_hits:1709
> keyspace_misses:291
> pubsub_channels:1
> pubsub_patterns:0
> latest_fork_usec:0
> migrate_cached_sockets:0
> slave_expires_tracked_keys:0
> active_defrag_hits:0
> active_defrag_misses:0
> active_defrag_key_hits:0
> active_defrag_key_misses:0
> 
> # Replication
> role:master
> connected_slaves:0
> master_replid:26c49e75472c5cbec827933f8c2a5a7cbd29a71e
> master_replid2:0000000000000000000000000000000000000000
> master_repl_offset:0
> second_repl_offset:-1
> repl_backlog_active:0
> repl_backlog_size:1048576
> repl_backlog_first_byte_offset:0
> repl_backlog_histlen:0
> 
> # CPU
> used_cpu_sys:0.09
> used_cpu_user:0.59
> used_cpu_sys_children:0.00
> used_cpu_user_children:0.00
> 
> # Commandstats
> cmdstat_hgetall:calls=1,usec=2,usec_per_call=2.00
> cmdstat_select:calls=19,usec=9,usec_per_call=0.47
> cmdstat_ping:calls=7,usec=4,usec_per_call=0.57
> cmdstat_info:calls=27,usec=715,usec_per_call=26.48
> cmdstat_echo:calls=10,usec=7,usec_per_call=0.70
> cmdstat_cluster:calls=5,usec=11,usec_per_call=2.20
> cmdstat_FT.AGGREGATE:calls=18,usec=860,usec_per_call=47.78
> cmdstat_script:calls=1,usec=103,usec_per_call=103.00
> cmdstat_config:calls=27,usec=208,usec_per_call=7.70
> cmdstat_hget:calls=1,usec=0,usec_per_call=0.00
> cmdstat_FT.SEARCH:calls=1,usec=148,usec_per_call=148.00
> cmdstat_scan:calls=9,usec=70511,usec_per_call=7834.56
> cmdstat_eval:calls=1,usec=38,usec_per_call=38.00
> cmdstat_client:calls=10,usec=16,usec_per_call=1.60
> cmdstat_subscribe:calls=11,usec=38,usec_per_call=3.45
> cmdstat_get:calls=9,usec=29,usec_per_call=3.22
> 
> # Cluster
> cluster_enabled:0
> 
> # Keyspace
> db0:keys=82576,expires=1,avg_ttl=85721485
> db1:keys=1,expires=0,avg_ttl=0
> db5:keys=750,expires=0,avg_ttl=0
> db6:keys=6600,expires=0,avg_ttl=0
> 
> ------ CLIENT LIST OUTPUT ------
> id=36 addr=0.0.0.0:54690 fd=10 name=HOSTNAME age=44 idle=44 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=info
> id=37 addr=0.0.0.0:54691 fd=11 name=HOSTNAME age=44 idle=44 flags=N db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=subscribe
> id=460 addr=0.0.0.0:59585 fd=17 name= age=10 idle=8 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=scan
> id=63 addr=0.0.0.0:54672 fd=12 name=HOSTNAME age=43 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=get
> id=64 addr=0.0.0.0:54673 fd=13 name=HOSTNAME age=43 idle=43 flags=N db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=subscribe
> id=25 addr=0.0.0.0:54687 fd=7 name=HOSTNAME age=44 idle=44 flags=N db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=subscribe
> id=392 addr=0.0.0.0:54694 fd=15 name=HOSTNAME age=19 idle=14 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=FT.AGGREGATE
> id=393 addr=0.0.0.0:54695 fd=16 name=HOSTNAME age=19 idle=19 flags=N db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=subscribe
> id=31 addr=0.0.0.0:54688 fd=8 name=HOSTNAME age=44 idle=0 flags=N db=2 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=eval
> id=68 addr=0.0.0.0:54674 fd=14 name=HOSTNAME age=43 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=0 oll=0 omem=0 events=r cmd=FT.SEARCH
> id=32 addr=0.0.0.0:54689 fd=9 name=HOSTNAME age=44 idle=44 flags=N db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=subscribe
> 
> ------ CURRENT CLIENT INFO ------
> id=68 addr=0.0.0.0:54674 fd=14 name=HOSTNAME age=43 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=0 oll=0 omem=0 events=r cmd=FT.SEARCH
> argv[0]: 'FT.SEARCH'
> argv[1]: 'idxCachedAuctionItem'
> argv[2]: '((((((@BatchStartDate:[-inf 1550674445] @BatchEndDate:[1550674445 +inf]) (@StatusId:[2 2]|@StatusId:[50 50]))|((((@BatchStartDate:[-inf 1550674445] @BatchEndDate:[-inf 1550674445]) @WillBeExtended:[1 1]) @IsExtendedPhaseFinished:[0 0]) (@StatusId:[2 2]|@StatusId:[50 50]))) (-@RestrictionCountryCodesInclude:[0 0] -@RestrictionCountryCodesInclude:[1 1])) ((-@RestrictionCompanyStatusIdsInclude:[0 0] -@RestrictionCompanyStatusIdsInclude:[1 1])|(@RestrictionCompanyStatusIdsInclude:[1 1] @RestrictionCompanyStatusIds:{0})|(@RestrictionCompanyStatusIdsInclude:[0 0] -@RestrictionCompanyStatusIds:{0}))) ((-@RestrictionCompanyIdsInclude:[0 0] -@RestrictionCompanyIdsInclude:[1 1])|(@RestrictionCompanyIdsInclude:[1 1] @RestrictionCompanyIds:{0})|(@RestrictionCompanyIdsInclude:[0 0] -@RestrictionCompanyIds:{0})))'
> argv[3]: 'RETURN'
> argv[4]: '1'
> argv[5]: 'AuctionId'
> argv[6]: 'LIMIT'
> argv[7]: '0'
> argv[8]: '999999'
> 
> ------ REGISTERS ------
> 508:M 20 Feb 14:54:05.472 # 
> RAX:0000000000000000 RBX:00007f46f7bfe800
> RCX:000055a6303b696c RDX:000055a62f3316c0
> RDI:00000000000000e1 RSI:00007f46f7a00000
> RBP:00007f46f7bff280 RSP:00007ffd04bd9f20
> R8 :0000000000000007 R9 :0000000000000007
> R10:000055a62f3318c0 R11:0000000000000020
> R12:0000000000000001 R13:000055a6303b696c
> R14:00007f471821ed60 R15:0000000000000002
> RIP:00007f4718223e7d EFL:0000000000010287
> CSGSFS:002b000000000033
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f2f) -> 000055a6303b5490
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f2e) -> 0000000000000010
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f2d) -> 00007ffd04bda0d0
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f2c) -> 000055a62f7cc460
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f2b) -> 3ff0000000000000
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f2a) -> 0000000000000004
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f29) -> 00007f4718220c02
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f28) -> 0000000000000002
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f27) -> 0000000000000000
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f26) -> 0000000000000000
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f25) -> 000055a6303b6e50
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f24) -> 000055a6303b6e50
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f23) -> 00007f471821fc56
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f22) -> 000055a6303b6850
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f21) -> 00007f46f99fd8a0
> 508:M 20 Feb 14:54:05.472 # (00007ffd04bd9f20) -> 0000000000120830
> 
> ------ FAST MEMORY TEST ------
> 508:M 20 Feb 14:54:05.472 # Bio thread for job type #0 terminated
> 508:M 20 Feb 14:54:05.472 # Bio thread for job type #1 terminated
> 508:M 20 Feb 14:54:05.472 # Bio thread for job type #2 terminated
> *** Preparing to test memory region 55a62f566000 (98304 bytes)
> *** Preparing to test memory region 55a62f7bc000 (135168 bytes)
> *** Preparing to test memory region 55a62f7dd000 (14598144 bytes)
> *** Preparing to test memory region 7f46e8000000 (135168 bytes)
> *** Preparing to test memory region 7f46ec000000 (135168 bytes)
> *** Preparing to test memory region 7f46f0000000 (135168 bytes)
> *** Preparing to test memory region 7f46f7800000 (276824064 bytes)
> *** Preparing to test memory region 7f4708200000 (10485760 bytes)
> *** Preparing to test memory region 7f4708e00000 (44040192 bytes)
> *** Preparing to test memory region 7f470ba00000 (23068672 bytes)
> *** Preparing to test memory region 7f470d1bb000 (8388608 bytes)
> *** Preparing to test memory region 7f470d9bc000 (8388608 bytes)
> *** Preparing to test memory region 7f470e1bd000 (8388608 bytes)
> *** Preparing to test memory region 7f470e9be000 (8388608 bytes)
> *** Preparing to test memory region 7f470f1bf000 (8388608 bytes)
> *** Preparing to test memory region 7f470f9c0000 (8388608 bytes)
> *** Preparing to test memory region 7f47101c1000 (8388608 bytes)
> *** Preparing to test memory region 7f47109c2000 (8388608 bytes)
> *** Preparing to test memory region 7f47111c3000 (8388608 bytes)
> *** Preparing to test memory region 7f47119c4000 (8388608 bytes)
> *** Preparing to test memory region 7f47121c5000 (8388608 bytes)
> *** Preparing to test memory region 7f47129c6000 (8388608 bytes)
> *** Preparing to test memory region 7f47131c7000 (8388608 bytes)
> *** Preparing to test memory region 7f47139c8000 (8388608 bytes)
> *** Preparing to test memory region 7f47141c9000 (8388608 bytes)
> *** Preparing to test memory region 7f47149ca000 (8388608 bytes)
> *** Preparing to test memory region 7f47151cb000 (8388608 bytes)
> *** Preparing to test memory region 7f47159cc000 (8388608 bytes)
> *** Preparing to test memory region 7f47161cd000 (8388608 bytes)
> *** Preparing to test memory region 7f47169ce000 (8388608 bytes)
> *** Preparing to test memory region 7f47171cf000 (8388608 bytes)
> *** Preparing to test memory region 7f47179d0000 (8388608 bytes)
> *** Preparing to test memory region 7f47185fa000 (12288 bytes)
> *** Preparing to test memory region 7f47185fe000 (8388608 bytes)
> *** Preparing to test memory region 7f4718dff000 (8388608 bytes)
> *** Preparing to test memory region 7f4719600000 (8388608 bytes)
> *** Preparing to test memory region 7f4719e00000 (4194304 bytes)
> *** Preparing to test memory region 7f471a5e8000 (16384 bytes)
> *** Preparing to test memory region 7f471a805000 (16384 bytes)
> *** Preparing to test memory region 7f471af19000 (16384 bytes)
> *** Preparing to test memory region 7f471af36000 (4096 bytes)
> .O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O
> Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.
> 
> ------ DUMPING CODE AROUND EIP ------
> Symbol: IR_GetCriteriaTester (base: 0x7f4718223e30)
> Module: /home/redisuser01/RediSearch/build/redisearch.so (base 0x7f47181d0000)
> $ xxd -r -p /tmp/dump.hex /tmp/dump.bin
> $ objdump --adjust-vma=0x7f4718223e30 -D -b binary -m i386:x86-64 /tmp/dump.bin
> ------
> 508:M 20 Feb 14:54:07.494 # dump of function (hexdump of 205 bytes):
> 488b074885c00f84e40000004883b828010000000f84d600000055534889fdbf400000004883ec08488b0521fb3c00ff104889c3488b450048894330488d05fd040000483945407457488b4558488b4028488b38488b0555f63c00ff1048894310488b4558488b4028488b400848894318488b4530488b553848894320488d050c08000048895328488903488d053effffff488943084883c4084889d85b5dc3488b4530488b3848897b10488b500848895318488b501048895320488b401848894328488b05e6f53c00ff1048
> 
> === REDIS BUG REPORT END. Make sure to include from START to END. ===
> 
>        Please report the crash by opening an issue on github:
> 
>            http://github.com/antirez/redis/issues
> 
>   Suspect RAM error? Use redis-server --test-memory to verify it.
```

**Comment by mnunberg** [comment]
@dme-development Is this reliably reproducible? I've just looked at this again and I can't help but conclude that this is a `malloc` failure- however, the amount of memory being allocated should be small, and nothing in the output suggests any kind of OOM condition.

Is this the first command that's being executed after startup? Can you also supply the exact redismodule.so that you are using? Maybe the symbol offsets in yours are a bit different and will point me elsewhere. Things like core dumps and/or gdb traces will also be useful here.

**Comment by mnunberg** [comment]
So it does not seem to be a malloc failure currently; however the code path itself should generally not happen. In any event, here is the assembly dump of the relevant crash. The crash happens at the final instruction:

```
00000000000540f0 <IR_GetCriteriaTester>:
   540f0:       48 8b 07                mov    (%rdi),%rax
   540f3:       48 85 c0                test   %rax,%rax
   540f6:       0f 84 e4 00 00 00       je     541e0 <IR_GetCriteriaTester+0xf0>
   540fc:       48 83 b8 28 01 00 00    cmpq   $0x0,0x128(%rax)
   54103:       00 
   54104:       0f 84 d6 00 00 00       je     541e0 <IR_GetCriteriaTester+0xf0>
   5410a:       55                      push   %rbp
   5410b:       53                      push   %rbx
   5410c:       48 89 fd                mov    %rdi,%rbp
   5410f:       bf 40 00 00 00          mov    $0x40,%edi
   54114:       48 83 ec 08             sub    $0x8,%rsp
   54118:       48 8b 05 29 08 3d 00    mov    0x3d0829(%rip),%rax        # 424948 <RedisModule_Alloc@@Base-0x6870>
   5411f:       ff 10                   callq  *(%rax)
   54121:       48 89 c3                mov    %rax,%rbx
   54124:       48 8b 45 00             mov    0x0(%rbp),%rax
   54128:       48 89 43 30             mov    %rax,0x30(%rbx)
   5412c:       48 8d 05 fd 04 00 00    lea    0x4fd(%rip),%rax        # 54630 <readNumeric>
   54133:       48 39 45 40             cmp    %rax,0x40(%rbp)
   54137:       74 57                   je     54190 <IR_GetCriteriaTester+0xa0>
   54139:       48 8b 45 58             mov    0x58(%rbp),%rax
   5413d:       48 8b 40 28             mov    0x28(%rax),%rax
```

**Comment by mnunberg** [comment]
However, the thing is that this function should never continue beyond the first few instructions in the first place...

**Comment by dme-development** [comment]
@mnunberg To answer your questions: it was not the first command executed after startup. Others where working. I even splitted up the FT.SEARCH command (before and after the first | operator). When executed both seperatly everything worked fine, when executed combined (as in the crash log), the crash happened.

I'm affraid I don't have the redismodule.so, I did a git pull and compiled it on the day before I created the issue, but now all our instances are running the tags/v1.4.3 version. I'm affraid I cannot supply any more info at this moment.
